treeherder/log_parser/parsers.py

```diff
@@ -422,6 +426,15 @@ def add(self, line, lineno):
 
     def parse_line(self, line, lineno):
         """Check a single line for an error.  Keeps track of the linenumber"""
+        # First line of TaskCluster logs almost certainly has this.
+        if line.startswith('[taskcluster '):
+            self.is_taskcluster = True
+
+        # For performance reasons, only do this if we have identified as
+        # a TC task.
```

Comment:
It seems to me like the check above for `if line.startswith('[taskcluster'):` would be about as much of a bottleneck? I don't know.

It's actually not hard to profile the log parser. Just download a log file into the root directory of your treeherder checkout and run the following inside the vagrant environment: 

```
time ./manage.py test_parse_log file:///home/vagrant/treeherder/my_log_file.txt.gz
```

Please give that a try-- if you notice a substantial difference, then we can leave this optimization in. Otherwise I'd prefer to keep things as simple as possible.